### PR TITLE
Add new command to extract issue resolved in a milestone

### PR DESCRIPTION
**Pull request by Amoifr**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Add new command to extract issues reslved in a milestone
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes 1057
| How to test?      | try the command with generated resume in the past milestones
| Possible impacts? | no

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
         "symfony/dotenv": "^4.3",
         "cache/filesystem-adapter": "^1.0",
         "symfony/yaml": "^5.2",
-        "justinrainbow/json-schema": "^5.2"
+        "justinrainbow/json-schema": "^5.2",
+        "symfony/var-dumper": "^5.4"
```
Comment (matks) [inline composer.json:12]: Maybe should be in require-dev

File: src/App/Service/Github/Filters.php [hunk 2]
```diff
@@ -9,6 +9,7 @@ class Filters
 
     public const FILTER_AUTHOR = 'author';
     public const FILTER_FILE_EXTENSION = 'fileExtension';
+    public const FILTER_MILESTONE = 'milestone';
```
Comment (matks) [inline src/App/Service/Github/Filters.php:12]: I dont see where this would be used

**Review by matks — COMMENTED** [review]
Great work! I added 2 suggestions

File: composer.json (hunk 1 above)
Comment (Amoifr) [reply composer.json]: yes ! my bad !

File: src/App/Service/Github/Filters.php (hunk 2 above)
Comment (Amoifr) [reply src/App/Service/Github/Filters.php]: ho yes ! a refactor later it's not used anymore. Thx !

File: README.md [hunk 3]
```diff
@@ -215,3 +215,54 @@ php bin/console slack:notifier:qa
 | `--ghtoken=<ghtoken>` | Yes/No | Use it or use .env |
 | `--slacktoken=<slacktoken>` | Yes/No | Use it or use .env |
 | `--slackchannel=<slackchannel>` | Yes/No | Use it or use .env |
+
+### Release Note helper
+Generate the list of the issue resolved in a milestone
```
Comment (Progi1984) [inline README.md:220]:
```suggestion
Generate the list of resolved issues in a milestone
```

File: README.md [hunk 4]
```diff
@@ -215,3 +215,54 @@ php bin/console slack:notifier:qa
...
+└─────────────────┴───────────────────────────────────────────────────────┴───────────────────────────────────────────────────────┘
+
+Output generated in 0s for 3 rows.
+```
+
+and
+
+
+```bash
+php bin/console github:release:note 1.7.8.2 --output=text
```
Comment (Progi1984) [inline README.md:255]:
```suggestion
php bin/console github:release:note 1.7.8.2 --output=markdown
```

File: src/App/Command/GithubReleaseNoteCommand.php [hunk 5]
```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Console\App\Command;
```
Comment (Progi1984) [inline src/App/Command/GithubReleaseNoteCommand.php:3]:
```suggestion
<?php

declare(strict_types=1);

namespace Console\App\Command;
```

File: src/App/Command/GithubReleaseNoteCommand.php [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    protected function configure()
+    {
+        $this->setName('github:release:note')
+            ->setDescription('Get release notes from github for a specific milestone')
+            ->addArgument('milestone', InputOption::VALUE_REQUIRED, 'Milestone to be checked')
+            ->addOption(
+                'output',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'output type (table|text)',
```
Comment (Progi1984) [inline src/App/Command/GithubReleaseNoteCommand.php:57]:
```suggestion
                'output type (table|markdown)',
```

File: src/App/Command/GithubReleaseNoteCommand.php [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+            $this->github->search($graphQLQuery)
+        );
+        $this->displayOutput($input, $milestone, $issues, $time);
+
+        return 0;
+    }
+
+    private function formatIssues(
+        array $resultAPI
+    ) {
```
Comment (Progi1984) [inline src/App/Command/GithubReleaseNoteCommand.php:84]:
```suggestion
    ): array {
```

File: src/App/Command/GithubReleaseNoteCommand.php [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+        $this->github = new Github($_ENV['GH_TOKEN']);
+        $this->output = $output;
+        $milestone = $input->getArgument('milestone');
+        if (!$milestone) {
+            $milestone = $this->io->ask('Please provide a milestone');
+        }
+        $time = time();
+
+        $graphQLQuery = new Query();
+        $graphQLQuery->setQuery('type:issue org:PrestaShop is:issue is:closed milestone:' . $milestone);
```
Comment (Progi1984) [inline src/App/Command/GithubReleaseNoteCommand.php:73]:
```suggestion
        $graphQLQuery->setQuery('type:issue repo:PrestaShop/PrestaShop is:issue is:closed milestone:' . $milestone);
```

File: README.md [hunk 9]
```diff
@@ -215,3 +215,54 @@ php bin/console slack:notifier:qa
...
+Generate the list of the issue resolved in a milestone
+
+```bash
+php bin/console github:release:note 
+```
+
+#### Parameters
+| Type                            | Name        | Required | Notes                                      |
+|---------------------------------|-------------|----------|--------------------------------------------|
+| Argument                        | milestone   | No       | milestone selected for search              |
```
Comment (Progi1984) [inline README.md:229]: Add an argument for the repository

File: src/App/Command/GithubReleaseNoteCommand.php (hunk 8 above)
Comment (Progi1984) [reply src/App/Command/GithubReleaseNoteCommand.php:73]: The repository should be a parameter.

**Review by Progi1984 — COMMENTED** [review]
Small feedback and a small improvement

File: src/App/Command/GithubReleaseNoteCommand.php [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+        $milestone = $input->getArgument('milestone');
+        if (!$milestone) {
+            $milestone = $this->io->ask('Please provide a milestone');
+        }
+        $time = time();
+
+        $graphQLQuery = new Query();
+        $graphQLQuery->setQuery(
+            sprintf(
+                'type:issue org:%s is:issue is:closed milestone:%s',
```
Comment (Progi1984) [inline src/App/Command/GithubReleaseNoteCommand.php:84]:
```suggestion
                'type:issue repo:PrestaShop/%s is:issue is:closed milestone:%s',
```

**Review by Progi1984 — COMMENTED** [review]
A small comment and it should be good

**Review by matks — DISMISSED** [review]
PR is globally OK we just need to polish a few things. I have nothing to add to @Progi1984 review but you need to fix the codestyle. PHP CS Fixer complains in the CI.

```
Run ./vendor/bin/php-cs-fixer fix --dry-run --diff
Loaded config PrestaShop coding standard from "/home/runner/work/presthubot/presthubot/.php_cs.dist".
   1) src/App/Command/GithubReleaseNoteCommand.php
      ---------- begin diff ----------
--- Original
+++ New
@@ @@
         array $resultAPI
-    ) :array {
+    ): array {
```

File: src/App/Command/GithubReleaseNoteCommand.php (hunk 10 above)
Comment (Amoifr) [reply src/App/Command/GithubReleaseNoteCommand.php:84]: applied ... also if I don't understand why ;-)

File: src/App/Command/GithubReleaseNoteCommand.php [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+        $milestone = $input->getArgument('milestone');
+        if (!$milestone) {
+            $milestone = $this->io->ask('Please provide a milestone');
+        }
+        $time = time();
+
+        $graphQLQuery = new Query();
+        $graphQLQuery->setQuery(
+            sprintf(
+                'type:issue repo:PrestaShop%s is:issue is:closed milestone:%s',
```
Comment (Progi1984) [inline src/App/Command/GithubReleaseNoteCommand.php:84]:
```suggestion
                'type:issue repo:PrestaShop/%s is:issue is:closed milestone:%s',
```

**Review by Progi1984 — CHANGES_REQUESTED** [review]
You forgot a character :)

File: src/App/Command/GithubReleaseNoteCommand.php (hunk 11 above)
Comment (Amoifr) [reply src/App/Command/GithubReleaseNoteCommand.php:84]: oups

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
I have tested this code locally and I confirm it works

I found a minor issue : my github token was expired but instead of being prompted an error message "Your GH token is expired" presthubot simply stayed frozen, I waited 1 minute before killing the process.

**Comment by matks** [comment]
Thank you @Amoifr

**Pull request merged**